plugins/node/instrumentation-amqplib/src/types.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Span } from '@opentelemetry/api';
+import { InstrumentationConfig } from '@opentelemetry/instrumentation';
+import type * as amqp from 'amqplib';
+
+export interface PublishInfo {
+  moduleVersion: string | undefined;
+  exchange: string;
+  routingKey: string;
+  content: Buffer;
+  options?: amqp.Options.Publish;
+  isConfirmChannel?: boolean;
+}
+
+export interface PublishConfirmedInfo extends PublishInfo {
+  confirmError?: any;
```

Comment:
```suggestion
  confirmError?: unknown;
```

Reply:
I am using the type as [published by `@types/amqplib`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/ea0e0de908b17dd697e373da9cd0fca8d7e41c66/types/amqplib/callback_api.d.ts#L54):

```js
    publish(exchange: string, routingKey: string, content: Buffer, options?: Options.Publish, callback?: (err: any, ok: Replies.Empty) => void): boolean;
```

Should this be changed to `unknown` only in instrumentation?

Reply:
`unknown` is more save. But I think it's ok stay inline with the original definitions.
But we should be aware that changing `any` to something else is usually breaking for some users and therefore most likely semver major.